mantle/platform/api/gcloud/compute.go

```diff
@@ -33,8 +34,56 @@ func (a *API) vmname() string {
 	return fmt.Sprintf("%s-%x", a.options.BaseName, b)
 }
 
+// ["local-ssd:interface=NVME,size=375"]
+func ParseDiskSpec(spec string, zone string) (*compute.AttachedDisk, error) {
+	split := strings.Split(spec, ":")
+	var disktype string
+	var diskinterface string
+	var disksize string
+	var size int64
+	var err error
+	disk_ps := "PERSISTENT"
+	if len(split) == 1 {
+		disktype = split[0]
+	} else if len(split) == 2 {
+		disktype = split[0]
+		for _, opt := range strings.Split(split[1], ",") {
+			if strings.HasPrefix(opt, "interface=") {
+				diskinterface = strings.TrimPrefix(opt, "interface=")
+				if diskinterface == "" || (diskinterface != "NVME" && diskinterface != "SCSI"){
+					return nil, fmt.Errorf("invalid interface opt %s", opt)
+				}
+			} else if strings.HasPrefix(opt, "size=") {
+				disksize = strings.TrimPrefix(opt, "size=")
+				size, err = strconv.ParseInt(disksize, 10, 32)
+				if err != nil {
+					return nil, fmt.Errorf("invalid size opt %s", opt)
```

Comment:
In Go the best practice here is to ["wrap" the original error](https://go.dev/blog/go1.13-errors), like:
```suggestion
					return nil, fmt.Errorf("invalid size opt %s: %w", opt, err)
```